sdk/storage/Azure.Storage.Common/tests/Shared/RetryOnFailedRequestAttribute.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+using NUnit.Framework.Internal.Commands;
+
+namespace Azure.Storage.Tests.Shared
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    public class RetryOnFailedRequestAttribute : NUnitAttribute, IRepeatTest
```

Comment:
borrowed from https://github.com/nunit/nunit/blob/b34eba3ac1aa6957157857bddd116256c634afab/src/NUnitFramework/framework/Attributes/RetryAttribute.cs#L17 .

Reply:
I'm afraid of attributes like this :) Can we just have the `RetryOnAuthorizationPermissionMismatch` and avoid giving people a gun?